graphx/src/main/scala/org/apache/spark/graphx/lib/PageRank.scala

```diff
@@ -188,7 +188,7 @@ object PageRank extends Logging {
       graph: Graph[VD, ED],
       numIter: Int,
       resetProb: Double = 0.15,
-      sources: Array[VertexId]): Graph[Vector, Double] = {
+      sources: Array[(VertexId, Double)]): Graph[Vector, Double] = {
```

Comment:
We can't change this API. It's possible to add another method and have both call into the same private helper function. What's an example of a use case for this?

Reply:
@srowen That sounds like a good idea too. Or should we just use an optional array parameter which has the scores?
We had an use case where we wanted to run some inferences on the graph and wanted the control on initial values of few of the nodes while running this algorithm.

Reply:
That also changes the binary signature, I believe. It's easy to avoid with a second method. If you can add a quick test of the new method and mark as `@Since(3.0.0)` I think this is probably OK, even though GraphX isn't really maintained anymore.

Reply:
Okay! I shall add the new methods then. 
So is there a naming convention followed for the method names? If so can you suggest me the names for the private method and for the new method which takes sources as Array[(VertexId, Double)]) 
Also it would be great if you can point me to the file containing test cases for these.

PS: What exactly do you mean by "GraphX isn't really maintained anymore"?

Reply:
Same public method name, maybe calling a "doRunParallel" method that's private.
There is really no new development on GraphX or MLlib; I wouldn't call them deprecated but is a reason to not add new methods to it. That said, this is minor enough that I think it's reasonable.